app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt

```diff
@@ -21,31 +39,117 @@ enum class Route(val titleStringId: Int, val route: String, val icon: Composable
     Home(R.string.navigation_home, "home", { Icon(Icons.Sharp.Home, null) }),
     Explore(R.string.navigation_explore, "explore", { Icon(Icons.Sharp.Search, null) }),
     Create(R.string.navigation_create, "create", { Icon(Icons.Sharp.Add, null) }),
-    ActiveHunts(R.string.navigation_active_hunts, "active-hunts", { Icon(androidx.compose.ui.res.painterResource(
-        id = R.drawable.target_arrow
-    ), null) }),
+    ActiveHunts(R.string.navigation_active_hunts, "active-hunts", {
+        Icon(
+            androidx.compose.ui.res.painterResource(
+                id = R.drawable.target_arrow
+            ), null
+        )
+    }),
     Profile(R.string.navigation_profile, "profile", { Icon(Icons.Sharp.Person, null) })
 
 }
 
 @Composable
-fun NavigationController(navController: NavHostController, modifier: Modifier = Modifier) {
+fun NavigationController(
+    navController: NavHostController,
+    database: Database,
+    modifier: Modifier = Modifier
+) {
+    val context = LocalContext.current
+    val authenticator = Authenticator.authInstance.get()
+    val activity: ComponentActivity = LocalContext.current.findActivity() as ComponentActivity
+
     NavHost(navController, startDestination = Route.Home.route, modifier = modifier) {
         composable(Route.Home.route) {
-            Text("Home")
+            Button(onClick = {
+                authenticator.signOut(activity)
+            }) {
+                Text("Sign out")
+            }
         }
         composable(Route.Explore.route) {
-            Text("Explore")
+            // Text("Explore")
+            Button(onClick = {
+                    navController.navigate("challenge-view/98d755ad-NQfISf2o_QzWLjCpedB")
+            }) {
+                Text(text = "Open challenge view")
+            }
         }
         composable(Route.Create.route) {
-            Text("Create")
+            CreateNewChallenge(
+                database = database,
+                onChallengeCreated = { challenge ->
+                    navController.popBackStack()
+                    navController.navigate("challenge-view/${challenge.cid}")
+                },
+                onFailure = {
+                    Toast.makeText(context, "Something went wrong, failed to create challenge", Toast.LENGTH_LONG).show()
+                    navController.popBackStack()
+                }
+            )
         }
         composable(Route.ActiveHunts.route) {
-            Text("Active hunts")
+            val user = Authenticator.authInstance.get().user
+
+            if(user == null) {
+                Text("Something went terribly wrong :/")
+            }
```

Comment:
I believe this should crash the application because this should be handle above (by @Maeeen)